src/clr-angular/data/datagrid/render/main-renderer.ts

```diff
@@ -92,6 +92,12 @@ export class DatagridMainRenderer<T = any> implements AfterContentInit, AfterVie
         this.setupColumns();
         this.columnsSizesStable = false;
         this.stabilizeColumns();
+      }),
+      this.columnsService.checkFirstVisible.subscribe(() => {
```

Comment:
nit: 
```ts
this.columnsService.checkFirstVisible.subscribe(this.findFirstVisible);
this.columnsService.checkLastVisible.subscribe(this.findLastVisible);
```

Reply:
We shouldn't do that. By referencing a function without invoking it, we would lose the correct context of this. That's one of the common red flags: https://github.com/microsoft/TypeScript/wiki/'this'-in-TypeScript